apps/builder/app/builder/features/style-panel/shared/style-info.ts

```diff
@@ -608,6 +605,26 @@ export const useStyleInfo = () => {
         defaultValue;
       if (value) {
         if (property === "color") {
+          const ownValue =
+            local ??
+            nextSource?.value ??
+            previousSource?.value ??
+            cascaded?.value ??
+            preset ??
+            htmlValue;
+          const ownColor =
+            ownValue?.type === "keyword" &&
+            (ownValue.value === "inherit" || ownValue.value === "currentColor")
+              ? undefined
+              : ownValue;
+          const inheritedValue = inherited?.value;
+          const inheritedColor =
+            inheritedValue?.type === "keyword" &&
+            (inheritedValue.value === "inherit" ||
+              inheritedValue.value === "currentColor")
+              ? undefined
+              : inheritedValue;
+          const currentColor = ownColor ?? inheritedColor ?? defaultValue;
```

Comment:
Not covered with tests at all
If I set for example this
`const inheritedColor = inheritedValue;`
tests works.

Having that logic is not that direct, I would add tests for each `if` above and add comments